snapcraft/internal/lxd.py

```diff
@@ -72,11 +72,14 @@ def __init__(self, *, output, source, project_options,
             kernel = server_environment['kernel_architecture']
         except KeyError:
             kernel = server_environment['kernelarchitecture']
-        deb_arch = _get_deb_arch(kernel)
-        if not deb_arch:
-            raise ContainerConnectionError(
-                'Unrecognized server architecture {}'.format(kernel))
-        self._image = 'ubuntu:xenial/{}'.format(deb_arch)
```

Comment:
Imho, old code is fine and should be tweaked slightly. Make ubuntu/xenial the default value for image and subtitue it in:
```
def __init__(self, ..., image='ubuntu:xenial'):
...
self._image = '{}/{}'.format(image, deb_arch)
```

And that's it. This way existing architecture handling is preserved but one can override the image to be "ubuntu:a" or "ubuntu-daily:b" etc.